juju/main.py

```diff
@@ -73,8 +73,12 @@ def _setup_hooks(charm_dir, charm):
     charm_dir -- A root directory of the charm.
     charm -- An instance of the Charm class.
     """
-    for event_name in charm.on.events().keys():
-        if event_name != 'install':
+    import juju.charm
+
+    for event_name, bound_event in charm.on.events().items():
+        is_juju_event = issubclass(bound_event.event_type, juju.charm.JujuEvent)
```

Comment:
I would have a `juju_events` function on `CharmEvents` and also move the `is_install_event` check to `_handle_event_link` because it already deals with some checking related to `"install"`:

```
for event_name, _ in charm.on.juju_events().items():
    _handle_event_link(charm_dir, event_name)
```

We do lose the type in that process and have to use a string comparison with event_kind though which doesn't seem very important in this case.

Reply:
Having a pre-filtered `juju_events` method on `CharmEvents` feels even more like exposing implementation ideas via the API than having a separate `JujuHookEvent` subclass.  It might also be worth noting that the `BoundEvent` has the info about the event name (`event_kind`) stored in it, so maybe we don't even need `CharmEvents.events()` to return a mapping and we can just pass the `bound_event` in instead.

Reply:
Moved to #19